lib/state_machine/state_machine.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "consts.hpp"
+#include "state.hpp"
+#include "types.hpp"
+
+#include <optional>
+#include <unordered_map>
+
+#include <boost/unordered_map.hpp>
+
+namespace hyped::state_machine {
+
+class StateMachine {
+ public:
+  StateMachine() { current_state = State::kIdle; };
```

Comment:
constructor should not be in the header, also initialise `current_state_` in the initialiser list